packages/react-i18n/src/utilities/translate.tsx

```diff
@@ -39,6 +45,83 @@ export function memoizedNumberFormatter(
   return i;
 }
 
+// Based on https://stackoverflow.com/a/68906367
+const stringNumberFormats = new Map<string, StringNumberFormatter>();
+export function memoizedStringNumberFormatter(
+  locales?: string | string[],
+  options?: Intl.NumberFormatOptions,
+): StringNumberFormatter {
+  const latnLocales = latinLocales(locales);
+  const key = numberFormatCacheKey(latnLocales, options);
+  if (stringNumberFormats.has(key)) {
+    return stringNumberFormats.get(key)!;
+  }
+
+  const formatter: StringNumberFormatter = {
+    format(amount: string): string {
+      let decimal = new Decimal(amount);
+      const originalSd = decimal.sd(false);
+
+      if (
+        options?.minimumSignificantDigits &&
+        originalSd > options.minimumSignificantDigits
+      ) {
+        decimal = decimal.toSignificantDigits(options.minimumSignificantDigits);
+      }
+
+      if (
+        options?.maximumSignificantDigits &&
+        originalSd > options.maximumSignificantDigits
+      ) {
+        decimal = decimal.toSignificantDigits(options.maximumSignificantDigits);
+      }
+
+      const [mainString, decimalString] = decimal.valueOf().split('.');
+
+      const decimalFormat = new Intl.NumberFormat(latnLocales, {
+        localeMatcher: options?.localeMatcher,
+        minimumFractionDigits: options?.minimumFractionDigits,
+        maximumFractionDigits: options?.maximumFractionDigits,
+      });
+      const decimalFullString = `0.${decimalString}`;
+      const decimalFullNumber = Number.parseFloat(decimalFullString);
+      const decimalFullFinal = decimalFormat.format(decimalFullNumber);
+      const decimalFinal = decimalFullFinal.slice(1);
+
+      const mainFormat = new Intl.NumberFormat(latnLocales, {
+        localeMatcher: options?.localeMatcher,
+        useGrouping: options?.useGrouping,
+        minimumIntegerDigits: options?.minimumIntegerDigits,
+        minimumFractionDigits: 0,
+      });
+      let mainBigInt = BigInt(mainString);
```

Comment:
Android has a Target SDK annotation you can use on methods and classes that will only work on X version of Android or later. They don't prevent compilation, but they let the developer know what to expect.

The problem with lack of BigInt support is that I don't think there is a fallback, unless we roll our own implementation of `Intl.NumberFormat`.

I made this PR initially to replicate the behavior we had on mobile native with arbitrarily large numbers for currency fields, but it turns out the web admin actually limits the size of input numbers, so it probably makes more sense to do that in RN instead of this.